src/Orleans/Providers/IStorageProvider.cs

```diff
@@ -70,7 +70,7 @@ public interface IStorageProvider : IProvider
     /// <summary>
     /// Interface to be optionally implemented by storage providers to return richer exception details.
     /// </summary>
-    public interface IExceptionDecoder
+    public interface IStorageExceptionDecoder
```

Comment:
But it decodes HttpStatusCode and rest code, so it is NOT generic. Perhaps it can be renamed HTTP exception decoder? Does it really matter? I can put it back as `IExceptionDecoder`.